File: service/src/main/java/com/jb/udp/Enum/BurnerAlarm.java
package com.jb.udp.Enum;

import java.util.ArrayList;
import java.util.EnumSet;
import java.util.List;

public enum BurnerAlarm {
    /// Bit2     热电偶故障
    thermocouple((byte)0x04),
    /// Bit1     旋转电机故障
    rotateFan((byte) 0x02),
    /// Bit0     进料电机故障
    feedFan((byte) 0x01);

    byte rawValue;

    BurnerAlarm(byte rawValue) {
        this.rawValue = rawValue;
    }

    public static EnumSet<BurnerAlarm> get(byte rawValue) {
        EnumSet<BurnerAlarm> enumSet = EnumSet.noneOf(BurnerAlarm.class);

        for (BurnerAlarm status : BurnerAlarm.values()) {
            if ((rawValue & status.rawValue) == status.rawValue) {
                enumSet.add(status);
            }
        }
        return enumSet;
    }

    public static List<String> des(EnumSet<BurnerAlarm> statuses) {
        List<String> d = new ArrayList<>();

        if (statuses.contains(thermocouple)) {
            d.add("热电偶故障");
        }
        if (statuses.contains(rotateFan)) {
            d.add("旋转电机故障");
        }
        if (statuses.contains(feedFan)) {
            d.add("进料电机故障");
        }
        return d;
    }
}

File: direct/platforms/ios/WeexDemo/UDP/BurnerAlarm.swift
//
//  BurnerAlarm.swift
//  WeexDemo
//
//  Created by zzzl on 2019/2/13.
//  Copyright © 2019 zzzl. All rights reserved.
//

import Foundation

struct BurnerAlarm: OptionSet {
    var rawValue: UInt8
    /// Bit2     热电偶故障
    static let thermocouple = BurnerAlarm(rawValue: 0x04)
    /// Bit1     旋转电机故障
    static let rotateFan = BurnerAlarm(rawValue: 0x02)
    /// Bit0     进料电机故障
    static let feedFan = BurnerAlarm(rawValue: 0x01)
}

extension BurnerAlarm {
    func des() -> [String] {
        var d = [String]()
        
        if contains(.thermocouple) {
            d.append("热电偶故障")
        }
        if contains(.rotateFan) {
            d.append("旋转电机故障")
        }
        if contains(.feedFan) {
            d.append("进料电机故障")
        }
        return d
    }
}
